components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -2238,11 +2238,17 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     public async findPrebuilds(ctx: TraceContext, params: FindPrebuildsParams): Promise<PrebuildWithStatus[]> {
         traceAPIParams(ctx, { params });
 
-        const user = this.checkAndBlockUser("getPrebuilds");
+        const user = this.checkAndBlockUser("findPrebuilds");
         await this.guardProjectOperation(user, params.projectId, "get");
         return this.projectsService.findPrebuilds(params);
     }
 
+    public async getPrebuild(ctx: TraceContext, prebuildId: string): Promise<PrebuildWithStatus | undefined> {
+        traceAPIParams(ctx, { prebuildId });
+        this.checkUser("getPrebuild");
```

Comment:
One last - but important - thing: We need to guard access to a prebuild.

So far, we don't have an access guard for prebuilds specifically (I think we should have one).
As placeholder, we mostly used the [WorkspaceInstance one](https://github.com/gitpod-io/gitpod/blob/3389f9a91c2e9401c2e49d97cf5b7d241d4a37bf/components/server/src/workspace/gitpod-server-impl.ts#L1542-L1543), which is ok to re-use in the scope of this PR.

But we should create a follow-up issue to fix this. :+1:

Reply:
Fixed with https://github.com/gitpod-io/gitpod/pull/10696/commits/8f2b6c8be82dcff9c63830ef440e4f61f50464fc